github/actions/client.go

```diff
@@ -975,28 +975,41 @@ func (c *Client) getActionsServiceAdminConnection(ctx context.Context, rt *regis
 
 	c.logger.Info("getting Actions tenant URL and JWT", "registrationURL", req.URL.String())
 
-	resp, err := c.Do(req)
-	if err != nil {
-		return nil, err
-	}
-	defer resp.Body.Close()
+	retry := 0
+	for {
```

Comment:
This might be painful because this is unexported, but can we add a test for this?

Reply:
Added, good call!